File: algorithms/leetcode/valid_palindrome.py


def validPalindrome(s):
    s = [c.lower() for c in s if c.isalnum()]
    if len(s) == 0:
        return True
    if len(s) == 1:
        return True
    if len(s) == 2:
        return s[0] == s[1]
    left  = 0
    right = len(s) -1
    while left < right and s[left] == s[right]:
        print(s)
        print(left, right)
        left  += 1
        if left == right:
            return True
        right -= 1
    return left == right
if __name__ == "__main__":
    print(validPalindrome("a"))

File: algorithms/leetcode/valid_palindrome.rs
fn valid_palindrome(s: String) -> bool {
    let mut left = 0;
    let mut right = s.len() - 1;
    let my_s: &[u8] = s
        .chars()
        .filter(|c| c.is_whitespace())
        .cloned()
        .collect::<String>()
        .as_bytes();
    while left < right && my_s[left] == my_s[right] {
        left += 1;
        right -= 1;
    }
    return left == right;
}
fn main() {
    let res = valid_palindrome(String::from("A man, a plan, a canal: Panama"));
    println!("{}", res);
}
